include/engine/data_watchdog.hpp

```diff
@@ -82,7 +83,7 @@ class DataWatchdog
     std::thread watcher;
```

Comment:
I agree, this would be sensible in the future. I would rather add this in another PR however.